src/browserlib/extract-algorithms.mjs

```diff
@@ -662,5 +666,9 @@ export default function (spec, idToHeading = {}) {
     return [];
   }
   const algorithms = findAlgorithms(document);
-  return algorithms.map(algo => serializeAlgorithm(algo));
+  return algorithms.map(algo => serializeAlgorithm(algo))
+    // remove duplicate from multi-steps algorithms also detected by the
+    // short form algorithms finder (relying on them being later in the
+    // list)
+    .filter((algo, idx, arr) => arr.findIndex(al => !algo.html || al.html === algo.html) === idx);
```

Comment:
This assumes that different algorithms will have different `html` properties. That's not the case, that property may contain pretty generic prose. E.g., 
[`createMediaKeys()`](https://w3c.github.io/encrypted-media/#dom-mediakeysystemaccess-createmediakeys) and [`createSession()`](https://w3c.github.io/encrypted-media/#dom-mediakeys-createsession) both end up with "When this method is invoked, the user agent MUST run the following steps". This code would drop the second one as a duplicate of the first.

At a minimum, you might want to only filter when `algo.rationale === 'To <dfn>'`, which should indeed drop the algorithm if it has already been extracted otherwise.

If you do that, I suspect you don't need the differentiated sort, and may be able to move the filtering back into findAlgorithms.

Reply:
I haven't found a way to find equivalence of algorithms at the `findAlgorithms` stage because `.html` isn't populated at that time. Thanks for flagging the fact that `.html` equality isn't sufficient for filtering; rather than `.rationale`, I think I'll make it applicable only if the list of steps is empty (at which points `.html` is guaranteed to be the only meaningful content)

Reply:
done in [11ea7a0](https://github.com/w3c/reffy/pull/1614/commits/11ea7a0a2da7fda11ac159c31546b3d5b86f59bc)

Reply:
A clear improvement, thanks!